### PR TITLE
ReadTheDocs site, User Guide sections

### PR DESCRIPTION
**Pull request by parente**
For #517

I'll setup a temporary RTD site out of my fork/branch for easier
consumption when the docs are far enough along. At the moment, I
don't think they're worth reviewing yet, except perhaps the general
outline of the doc sections.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+docker run -d -p 8888:8888 jupyter/base-notebook start-notebook.sh --NotebookApp.base_url=/some/path
+```
+
+For example, to disable all authentication mechanisms (which is not a recommended practice):
+
+```
+docker run -d -p 8888:8888 jupyter/base-notebook start-notebook.sh --NotebookApp.token=''
+```
+
+You can sidestep the `start-notebook.sh` script and run your own commands in the container. See the *Alternative Commands* section later in this document for more information.
```
Comment (ericdill) [inline docs/configuration.md]: does linking within the same document work like ``[Alternative Commands](#alternative-commands)``

File: Makefile [hunk 2]
```diff
@@ -54,10 +54,13 @@ dev/%: PORT?=8888
 dev/%: ## run a foreground container for a stack
 	docker run -it --rm -p $(PORT):8888 $(DARGS) $(OWNER)/$(notdir $@) $(ARGS)
 
-test-reqs: # install libraries required to run the integration tests
-	pip install -r requirements-test.txt
+dev-env: ## install libraries required to build docs and run tests
```
Comment (parente) [inline Makefile]: Need to adjust travis config to use this. Update `.PHONY` too for good measure.

**Comment by parente** [comment]
Staging site for this PR now exists here for easier viewing:

http://jupyter-docker-stacks-dev.readthedocs.io/en/rtd/

**Comment by parente** [comment]
I think the User Guide section of this PR is far enough along to invite general review now. I think the best way forward is to get what exists reviewed and merged, point to the resulting RTD site from the various READMEs, update the new docs site with any changes currently inflight in other PRs, and write the Contributor Guide sections in follow-on PRs.

@jzf2101 @willingc @minrk @ericdill or anyone else that has an interest in docs, would you care to take a look?

Temporary staging site lives here http://jupyter-docker-stacks-dev.readthedocs.io/en/rtd/ until this PR merges.

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+pygments_style = 'sphinx'
+
+# If true, `todo` and `todoList` produce output, else they produce nothing.
+todo_include_todos = False
+
+
+# -- Source -------------------------------------------------------------
+
+source_parsers = {
+    '.md': 'recommonmark.parser.CommonMarkParser',
```
Comment (ericdill) [inline docs/conf.py]: does this mean i can use markdown with sphinx?

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
+Jupyter Docker Stacks
+=====================
+
+Jupyter Docker Stacks are a set of ready-to-run Docker images containing Jupyter applications and interactive computing tools. You can use a stack image to start a personal Jupyter Notebook server in a local Docker container, to run JupyterLab servers for a team using JupyterHub, to write your own project Dockerfile, and so on.
```
Comment (ericdill) [inline docs/index.rst]: thought: make this a bulleted list with links to *how* to do each of these things (if that makes sense)

File: docs/conf.py (hunk 3 above)
Comment (parente) [reply docs/conf.py]: Sure can!

Comment (ericdill) [reply docs/conf.py]: Mind. Blown.

Comment (willingc) [reply docs/conf.py]: Yes! You can also use notebooks with sphinx. ;-)

File: docs/using/common.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+docker run -d -p 8888:8888 jupyter/base-notebook start-notebook.sh --NotebookApp.password='sha1:74ba40f8a388:c913541b7ee99d15d5ed31d4226bf7838f83a50e'
+```
+
+For example, to set the base URL of the notebook server, you can run the following:
+
+```
+docker run -d -p 8888:8888 jupyter/base-notebook start-notebook.sh --NotebookApp.base_url=/some/path
+```
+
+For example, to ignore best practice and disable all authentication, you can run the following:
```
[the remark this answers is not in the document]
Comment (parente) [reply docs/using/common.md]: It was added to the READMEs originally due to repeat requests about how to do it. That was some time ago. It's probably a good idea to remove it and see if it surfaces again.

**Comment by parente** [comment]
@willingc that sounds fine to me. Let me get @ericdill 's few suggestions in this evening and then we can merge to iterate.

**Comment by parente** [comment]
Merged. Happy to see any changes you suggest @willingc. I'll send another PR to see the contributing sections.

**Comment by jzf2101** [comment]
@parente does this close #517 ?

**Comment by parente** [comment]
@jzf2101 thanks for checking. There’s more to do before we can close it I think. The contributing sections are empty, @willingc suggested we iterate a bit over the user guide sections, and we need to remove most of the READMEs and point to the RTD site when it’s live. 

If also like to seed a cookie cutter project for community stack repos as part of that ticket, and provided the necessary doc on how to use it in the contributor guide here.

**Pull request merged**